bpf/binary.go

```diff
@@ -44,7 +46,34 @@ func BinaryFromFile(ifile string) (*Binary, error) {
 
 // WriteToFile writes the binary to a file
 func (b *Binary) WriteToFile(ofile string) error {
-	return ioutil.WriteFile(ofile, b.raw, 0600)
+	err := ioutil.WriteFile(ofile, b.raw, 0600)
+	if err != nil {
+		return err
+	}
+
+	// Append a UUID to the file.  We want each attachment point to get its own jump map
+	// but tc names jump maps by hash of the binary, which means they can clash if we load
+	// the same binary onto multiple interfaces.
+	f, err := os.OpenFile(ofile, os.O_APPEND|os.O_WRONLY, 0600)
```

Comment:
nice! I guess it does not matter what rubbish follows the ELF, or does it?

Reply:
Yes, tc doesn't seem to care about extra data at the end.  I think it's pretty common to stuff data at the end of an executable and the ELF spec seems to allow it (e.g. an auto-extracting archive).